subsys/net/lib/openthread/platform/radio.c

```diff
@@ -303,6 +307,9 @@ void transmit_message(struct k_work *tx_job)
 
 static inline void handle_tx_done(otInstance *aInstance)
 {
+	sTransmitFrame.mInfo.mTxInfo.mIsSecurityProcessed =
+		net_pkt_ieee802154_frame_secured(tx_pkt);
+
```

Comment:
I thought it's necessary to set `.mIsARetx` based on `_mac_hdr_rdy()` here as well. Is it omitted intentionally?

Reply:
It's intentional. OpenThread already infers the value of this flag from the transmission result:
- For direct transmissions, in `SubMac::HandleTransmitDone`
- For indirect transmissions, in `DataPollHandler::HandleFrameRequest`
- For CSL transmissions, in `CslTxScheduler::HandleFrameRequest`

If `.mIsARetx` is set based on `_mac_hdr_rdy()` even successful transmissions will be marked as retransmissions and wrong behavior will occur, like a CSL Receiver sending a Data Poll to re-synchronize the CSL IEs with the Parent.

Reply:
But it would work incorrectly. OT expects that frame with `.mIsARetx` flag set has correct frame counter and CSL IE values set. However, if `_mac_hdr_rdy()` is `false`, then these fields in the frame have incorrect value. It causes inconsistent state of frame data and metadata in OpenThread MAC.

I'm looking at OpenThread implementation for possible solutions

Reply:
Solution provided in https://github.com/openthread/openthread/pull/6849